commands/phantom.js

```diff
@@ -0,0 +1,567 @@
+      // 'button:text("Import Selected Accounts")',
+    );
+
+    // STEP: Input password, confirm and continue
+    await new Promise(resolve => setTimeout(resolve, 1000)); // the transitioning is too fast
+    await playwright.waitAndType(
+      PROVIDER,
+      firstTimeFlowImportPageElements.passwordInput,
+      password,
+    );
+    await playwright.waitAndType(
+      PROVIDER,
+      firstTimeFlowImportPageElements.confirmPasswordInput,
+      password,
+    );
+    await playwright.waitAndClick(
+      PROVIDER,
+      firstTimeFlowImportPageElements.termsCheckbox,
+    );
+    // continue to next screen
+    await playwright.waitAndClick(
+      PROVIDER,
+      firstTimeFlowImportPageElements.continueAfterPasswordButton,
+    );
+    // finish
+    await new Promise(resolve => setTimeout(resolve, 1000)); // the transitioning is too fast
+    await playwright.waitAndClick(
+      PROVIDER,
+      firstTimeFlowImportPageElements.getStartedButton,
+    );
+    return true;
+  },
+  closePopupAndTooltips: async () => {
+    // note: this is required for fast execution of e2e tests to avoid flakiness
+    // otherwise popup may not be detected properly and not closed
+    await playwright.windows(PROVIDER).waitForTimeout(1000);
+    if (
+      await playwright
+        .windows(PROVIDER)
+        .locator(mainPageElements.popup.container)
+        .isVisible()
+    ) {
+      const popupBackground = playwright
+        .windows(PROVIDER)
+        .locator(mainPageElements.popup.background);
+      const popupBackgroundBox = await popupBackground.boundingBox();
+      await playwright
+        .windows(PROVIDER)
+        .mouse.click(popupBackgroundBox.x + 1, popupBackgroundBox.y + 1);
+    }
+    if (
+      await playwright
+        .windows(PROVIDER)
+        .locator(mainPageElements.tippyTooltip.closeButton)
+        .isVisible()
+    ) {
+      await playwright.waitAndClick(
+        PROVIDER,
+        mainPageElements.tippyTooltip.closeButton,
+      );
+    }
+    if (
+      await playwright
+        .windows(PROVIDER)
+        .locator(mainPageElements.actionableMessage.closeButton)
+        .isVisible()
+    ) {
+      await playwright.waitAndClick(
+        PROVIDER,
+        mainPageElements.actionableMessage.closeButton,
+      );
+    }
+    return true;
+  },
+  getWalletAddress: async (chain = 'eth') => {
+    await switchToPhantomIfNotActive();
+    await playwright.windows(PROVIDER).hover(mainPageElements.accountBar.title);
+    await new Promise(resolve => setTimeout(resolve, 100));
+
+    if (chain === 'eth') {
+      await playwright.waitAndClick(
+        PROVIDER,
+        mainPageElements.accountBar.ethRow,
+      );
+    } else if (chain === 'solana') {
+      await playwright.waitAndClick(
+        PROVIDER,
+        mainPageElements.accountBar.solanaRow,
+      );
+    }
+
+    walletAddress = await playwright
+      .windows(PROVIDER)
+      .evaluate('navigator.clipboard.readText()');
```

Comment:
I'd be very careful with using the raw `clipboard.readText()` function. Let's change this code to "emulate" `CTRL/CMD + V` shortcut.

Reply:
i found this to work rather well though. we're using the browser api so it should be fine, also correct permission was set